python/mmSolver/tools/removesolvernodes/tool.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2020 John Smith.
+#
+# This file is part of mmSolver.
+#
+# mmSolver is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# mmSolver is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with mmSolver.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""
+Remove Solver Nodes - user facing.
+"""
+from functools import wraps
+
+import mmSolver.logger
+import mmSolver.api as mmapi
+import maya.cmds as cmds
+import mmSolver._api.utils as api_utils
+
+LOG = mmSolver.logger.get_logger()
+
+def undoStack(func):
```

Comment:
Rather than using this undoStack undo wrapper, I already have this functionality built-in to mmSolver with `mmSolver.utils.tools.tool_context`, can you use that instead?

https://david-cattermole.github.io/mayaMatchMoveSolver/mmSolver.utils.html#mmSolver.utils.tools.tool_context